lib/openstudio-standards/standards/ashrae_90_1_prm/docs/set_baseline_wwr.md

```diff
@@ -0,0 +1,191 @@
+    - calculate total window area `wind_m2 = calculate_total_wind_area(space_group)`
+    - get the window to wall ratio (wwr) limit from Table 3.1.1-1 `wwr_lim = table_3.1.1-1(wwr_building_type)`
+    - calculate multiplier `multiplier = wwr_lim / (wind_m2 / wall_m2)`
+    - For `space` in `space_group`:
+      - For `surface` in `space.surfaces` where `surface_boundary_condition == 'Outdoor' && surfaceType=='Wall'`:
+        - set reduction ratio `reduction_ratio = 1.0` 
+        - calculate the window to wall ratio of the surface `surface_wwr = get_wwr_of_a_surface`
+        - if reduction is requested `multiplier < 1.0`
+          - reduction ratio is set to `1.0 - multiplier`
+          - call function to shrink the fenestration towards centroid `sub_surface_reduce_area_by_percent_by_shrinking_toward_centroid`
+        - elsif `wwr_building_type == 'all others'`
+          - reduction ratio set to `1.0`
+          - do nothing
+        - else
+          - calculate maximum fenestration area that wall surfaces with fenestration can reach `max_wwr = fene_only_wall_m2 * 0.9 / wall_m2` (Note: 90% wwr is the upper limit for each surface in OpenStudio standard)
+          - if the wwr limit is greater than max wwr `max_wwr < wwr_lim`
+            - if the surface has no fenestration `if surface_wwr == 0` (Note: In this case, it is required to add new fenestration to the surface)
+              - Add window to the surface by a calculated ratio `surface.setWindowToWallRatio(1.0 + (wwr_lim*wall_m2 - max_wwr*wall_m2)/(wall_m2-fene_only_wall_m2))`
+            - else: (Note: In this case, it is required to maximize the fenestration area in this surface to 90% wwr)
+              - Set window to wall ratio to 90% `surface.setWindowToWallRatio(0.9)`
+          else
+            - if the surface has no fenestration `if surface_wwr == 0`
+              - Do nothing
+            - else:
+              - Set a new window to wall ratio to the surface `surface.setWindowToWallRatio(surface_wwr * multiplier)`
+
+## Key Ruby Methods
+The function `get_wwr_reduction_ratio` implements the logic for calculating the reduction ratio `reduction_ratio` and `model_adjust_fenestration_in_a_surface` implements the logics to adjust the window to wall ratio for a surface
+
+The implementation of both functions can be found in the `ashrae_90_1_prm.Model.rb`
+
+## Notes:
+The function works well with cases where WWR reduction is needed. For WWR increase to meet Table G3.1.1-1 values, the function's limitation is listed below:
+1. The function will remove all windows in the surface then add a new window to meet the required WWR.
+2. The function will keep the existing doors but there could be cases the location of door overlaps with the new window.
+
+In addition, when increasing the WWR requires adding windows to surfaces with no windows, the function will not adding any windows to spaces tagged as plenum and used as a AirLoopSupplyPlenum or AirLoopReturnPlenum. 
```

Comment:
Should it be "to spaces tagged as plenum **and/or** used as a AirLoopSupplyPlenum or AirLoopReturnPlenum